utils.go

```diff
@@ -139,37 +145,40 @@ func toYAML(file string, s *state) {
 	newFile.Close()
 }
 
-// substituteEnvInValuesFiles loops through the values/secrets files and substitutes env variables into them
-func substituteEnvInValuesFiles(s *state) {
-	log.Println("INFO: substituting env variables in values and secrets files ...")
+// substituteVarsInValuesFiles loops through the values/secrets files and substitutes variables into them.
+func substituteVarsInValuesFiles(s *state) {
 	for _, v := range s.Apps {
 		if v.ValuesFile != "" {
-			v.ValuesFile = substituteEnvInYaml(v.ValuesFile)
+			v.ValuesFile = substituteVarsInYaml(v.ValuesFile)
 		}
 		if v.SecretsFile != "" {
-			v.SecretsFile = substituteEnvInYaml(v.SecretsFile)
+			v.SecretsFile = substituteVarsInYaml(v.SecretsFile)
 		}
 		for i := range v.ValuesFiles {
-			v.ValuesFiles[i] = substituteEnvInYaml(v.ValuesFiles[i])
+			v.ValuesFiles[i] = substituteVarsInYaml(v.ValuesFiles[i])
 		}
 		for i := range v.SecretsFiles {
-			v.SecretsFiles[i] = substituteEnvInYaml(v.SecretsFiles[i])
+			v.SecretsFiles[i] = substituteVarsInYaml(v.SecretsFiles[i])
 		}
 	}
 }
 
-// substituteEnvInYaml substitutes env variables in a Yaml file and creates a temp file with these values.
+// substituteVarsInYaml substitutes variables in a Yaml file and creates a temp file with these values.
 // Returns the path for the temp file
-func substituteEnvInYaml(file string) string {
+func substituteVarsInYaml(file string) string {
 	rawYamlFile, err := ioutil.ReadFile(file)
-	var yamlFile string
 	if err != nil {
 		logError(err.Error())
 	}
-	if !noEnvSubst {
-		yamlFile = substituteEnv(string(rawYamlFile))
-	} else {
-		yamlFile = string(rawYamlFile)
+
+	yamlFile := string(rawYamlFile)
+	if !noEnvSubst && !noEnvValuesSubst {
```

Comment:
I'm not sure this keeps the same functionality as before, is `substituteVarsInYaml` only used for the values files?

Reply:
Hi @luisdavim, thanks for looking at my MR! 

To answer your question, yes, `substituteVarsInYaml` is only ever called from `substituteVarsInValuesFiles` and therefore only used for the values files. 

With the introduction of the `noSSMSubst` and `noSSMValuesSubst` flags, it was necessary to push the conditional checks down the stack to a point where flags for SSM param evaluation and Env param evaluation could be checked independently without one precluding the other. I also didn't want to create a second set of temporary files in order to add the SSM param feature, so this was my best choice (at the time) for a home for this logic. 

Do you see any issues with this approach?